drivers/adc/adc_stm32.c

```diff
@@ -627,21 +625,13 @@ static void adc_stm32_oversampling(ADC_TypeDef *adc, uint8_t ratio, uint32_t shi
 {
 	adc_stm32_oversampling_scope(adc, LL_ADC_OVS_GRP_REGULAR_CONTINUED);
 #if defined(CONFIG_SOC_SERIES_STM32H7X)
-	/*
-	 * Set bits manually to circumvent bug in LL Libraries
-	 * https://github.com/STMicroelectronics/STM32CubeH7/issues/177
-	 */
-#if defined(ADC_VER_V5_V90)
-	if (adc == ADC3) {
-		MODIFY_REG(adc->CFGR2, (ADC_CFGR2_OVSS | ADC3_CFGR2_OVSR),
-			(shift | stm32_adc_ratio_table[ratio]));
+	if (adc != ADC3) {
+		/* the LL function expects a value from 1 to 1024 */
+		adc_stm32_oversampling_ratioshift(adc, 1 << ratio, shift);
 	} else {
-		MODIFY_REG(adc->CFGR2, (ADC_CFGR2_OVSS | ADC_CFGR2_OVSR),
-			(shift | (((1UL << ratio) - 1) << ADC_CFGR2_OVSR_Pos)));
+		/* the LL function expects a value LL_ADC_OVS_RATIO_x */
+		adc_stm32_oversampling_ratioshift(adc, stm32_adc_ratio_table[ratio], shift);
```

Comment:
Sorry, I haven't seen that in my first review, but you also need to check which version of ADC3 you have. Only the ADC3 of STM32H72x/H73x is impacted by this, for the rest of the H7 family ADC3 behaves like ADC1/2.
You need to have a `#ifdef ADC_VER_V5_V90` and proceed like in the `LL_ADC_ConfigOverSamplingRatioShift` in stm32h7xx_ll_adc.h.

Reply:
Ah, you're right. I wrongly assumed the fix would allow the call the be more generic. Should've caught that, my bad.